### PR TITLE
feat(windows): Expose webview2 zoom hotkeys from wry

### PR DESCRIPTION
**Pull request by Legend-Master**
Reference: #3310

### CODE REVIEW

File: core/tauri-utils/src/config.rs [hunk 1]
```diff
@@ -1276,6 +1276,13 @@ pub struct WindowConfig {
   ///
   /// - **macOS**: Requires the `macos-proxy` feature flag and only compiles for macOS 14+.
   pub proxy_url: Option<Url>,
+  /// Whether page zooming by hotkeys is enabled
+  ///
+  /// ## Platform-specific
+  ///
+  /// **macOS / Linux / Android / iOS**: Unsupported
```
Comment (amrbashir) [inline core/tauri-utils/src/config.rs]:
```suggestion
  /// Whether page zooming by hotkeys is enabled **Windows Only**
```

File: core/tauri/src/webview/mod.rs [hunk 2]
```diff
@@ -775,6 +775,17 @@ fn main() {
     self.webview_attributes.auto_resize = true;
     self
   }
+
+  /// Whether page zooming by hotkeys is enabled
+  ///
+  /// ## Platform-specific
+  ///
+  /// **macOS / Linux / Android / iOS**: Unsupported
```
Comment (amrbashir) [inline core/tauri/src/webview/mod.rs]:
```suggestion
  /// Whether page zooming by hotkeys is enabled **Windows Only**
```

File: core/tauri/src/webview/webview_window.rs [hunk 3]
```diff
@@ -838,6 +838,17 @@ fn main() {
     self.webview_builder = self.webview_builder.proxy_url(url);
     self
   }
+
+  /// Whether page zooming by hotkeys is enabled
+  ///
+  /// ## Platform-specific
+  ///
+  /// **macOS / Linux / Android / iOS**: Unsupported
```
Comment (amrbashir) [inline core/tauri/src/webview/webview_window.rs]:
```suggestion
  /// Whether page zooming by hotkeys is enabled **Windows only**
```

File: tooling/api/src/webview.ts [hunk 4]
```diff
@@ -648,6 +648,14 @@ interface WebviewOptions {
    * - **macOS**: Requires the `macos-proxy` feature flag and only compiles for macOS 14+.
    * */
   proxyUrl?: string
+  /**
+   * Whether page zooming by hotkeys or gestures is enabled
+   *
+   * ## Platform-specific
+   *
+   * - **macOS / Linux / Android / iOS**: Unsupported
```
Comment (amrbashir) [inline tooling/api/src/webview.ts]:
```suggestion
   * Whether page zooming by hotkeys or gestures is enabled **Windows Only**
```

**Review by amrbashir — CHANGES_REQUESTED** [review]
Just a few documentation nitpicks. You will need to regenerate the schema file.

**Comment by lucasfernog** [comment]
Can we use [WebView::zoom](https://docs.rs/wry/latest/wry/struct.WebView.html#method.zoom) to manually implement this feature on macOS/Linux?

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
This PR is for only allowing users to enable the builtin one on Webview2, we need another PR to expose [WebView::zoom](https://docs.rs/wry/latest/wry/struct.WebView.html#method.zoom) and users can create a UI for it themselves.

**Comment by amrbashir** [comment]
hmm actually this doesn't enable any UI, just the hotkeys, maybe we can indeed emulate the behavior on other platforms and also expose [WebView::zoom](https://docs.rs/wry/latest/wry/struct.WebView.html#method.zoom) at the same time

@Legend-Master would you like to add that too in this PR?

**Comment by Legend-Master** [comment]
I think it's fine to expose the webview2 one first in this pr (it's in the webview2 api but we don't have a way to use it), and expose zoom function and polyfill for this functionality in another pr

If we want to put them in a single pr, I can take a look tomorrow, but I'm not familiar with WebKit zoom controls (is it the same as Chromium?)

And emulating it would be injecting some JavaScript to the webview?

**Comment by amrbashir** [comment]
> If we want to put them in a single pr, I can take a look tomorrow

let's make them in another PR.

> but I'm not familiar with WebKit zoom controls (is it the same as Chromium?)

Yeah should be the same keys, `ctrl +/-`.

> And emulating it would be injecting some JavaScript to the webview?

yeah, take a look at https://github.com/tauri-apps/tauri/blob/36b4c12497fbe636066f4848c6877b3ab6cc892e/core/tauri/src/webview/scripts/toggle-devtools.js

**Comment by Legend-Master** [comment]
Seems like webview2's zoom popup is disabled without a way to getting it back https://github.com/MicrosoftEdge/WebView2Feedback/issues/95#issuecomment-591590945, also the zoom factor is not persisted across page load (reset to 100% on reload/restart), pretty sad that webview2's zoom control is a downgraded version of the browser zoom control 😥

I'll take a look at exposing `zoom` function now

**Pull request merged**